tests/ixia/pfcwd/files/pfcwd_basic_helper.py

```diff
@@ -0,0 +1,301 @@
+        pfc_storm_dur_sec = ceil(detect_time_sec + poll_interval_sec + 0.1)
+
+        data_flow1_delay_sec = restore_time_sec / 2
+        data_flow1_dur_sec = pfc_storm_dur_sec
+
+        """ Start data traffic 2 after PFC is restored """
+        data_flow2_delay_sec = pfc_storm_dur_sec + restore_time_sec + poll_interval_sec
+        data_flow2_dur_sec = 1
+
+        data_flow1_max_loss_rate = 1
+        data_flow1_min_loss_rate = 1- DEVIATION
+
+    else:
+        pfc_storm_dur_sec = detect_time_sec * 0.5
+        data_flow1_delay_sec = pfc_storm_dur_sec * 0.1
+        data_flow1_dur_sec = ceil(pfc_storm_dur_sec)
+
+        """ Start data traffic 2 after the completion of data traffic 1 """
+        data_flow2_delay_sec = data_flow1_delay_sec + data_flow1_dur_sec + 0.1
+        data_flow2_dur_sec = 1
+
+        data_flow1_max_loss_rate = 0
+        data_flow1_min_loss_rate = 0
+
+    exp_dur_sec = data_flow2_delay_sec + data_flow2_dur_sec + 1
+
+    """ Generate traffic config """
+    flows = __gen_traffic(testbed_config=testbed_config,
+                          port_id=port_id,
+                          pause_flow_name=PAUSE_FLOW_NAME,
+                          pause_flow_dur_sec=pfc_storm_dur_sec,
+                          data_flow1_name=DATA_FLOW1_NAME,
+                          data_flow1_delay_sec=data_flow1_delay_sec,
+                          data_flow1_dur_sec=data_flow1_dur_sec,
+                          data_flow2_name=DATA_FLOW2_NAME,
+                          data_flow2_delay_sec=data_flow2_delay_sec,
+                          data_flow2_dur_sec=data_flow2_dur_sec,
+                          data_pkt_size=DATA_PKT_SIZE,
+                          prio_list=prio_list,
+                          prio_dscp_map=prio_dscp_map)
+
+    """ Tgen config = testbed config + flow config """
+    config = testbed_config
+    config.flows = flows
+
+    all_flow_names = [flow.name for flow in flows]
+
+    flow_stats = __run_traffic(api=api,
+                               config=config,
+                               all_flow_names=all_flow_names,
+                               exp_dur_sec=exp_dur_sec)
+
+    __verify_results(rows=flow_stats,
+                     data_flow1_name=DATA_FLOW1_NAME,
+                     data_flow1_min_loss_rate=data_flow1_min_loss_rate,
+                     data_flow1_max_loss_rate=data_flow1_max_loss_rate,
+                     data_flow2_name=DATA_FLOW2_NAME,
+                     data_flow2_min_loss_rate=0,
+                     data_flow2_max_loss_rate=0)
+
+sec_to_nanosec = lambda x : x * 1e9
+
+def __gen_traffic(testbed_config,
+                  port_id,
+                  pause_flow_name,
+                  pause_flow_dur_sec,
+                  data_flow1_name,
+                  data_flow1_delay_sec,
+                  data_flow1_dur_sec,
+                  data_flow2_name,
+                  data_flow2_delay_sec,
+                  data_flow2_dur_sec,
```

Comment:
These can be passed in as a list [(flow1_name, flow1_delay_sec, flow1_dur, sec), ...]. This gives the flexibility to pass in 'n' number of flows